### PR TITLE
WL-3453 Don't allow sysadmin to bypass security checks.

### PR DESCRIPTION
**Pull request by buckett**
Doing good tests requires too much refactoring so not including that work.

### CODE REVIEW

File: kernel-impl/src/test/java/org/sakaiproject/authz/impl/TwoFactorAuthenticationIntTest.java [hunk 1]
```diff
@@ -108,6 +108,14 @@ public void testTwoFactorRequired() throws Exception {
 
 		assertTrue(twoFactorAuthentication.isTwoFactorRequired(secureSite.getReference()));
```
Comment (nicholaswilson100) [inline kernel-impl/src/test/java/org/sakaiproject/authz/impl/TwoFactorAuthenticationIntTest.java:110]: Note: I think it's really cool to be able to declare the expectation that a particular excpetion will be thrown in a test in annotations like below.  You can even test for the exception message. Not sure if it's set up to do that here though...  

@Test(expected = RuntimeException.class)
public void testForExpectedExceptionWithAnnotation()
        throws Exception {
    divide (1, 0);
}

File: kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java [hunk 2]
```diff
@@ -2552,45 +2494,57 @@ public boolean allowRemoveCollection(String id)
...
 		{
 			ResourceProperties props = getProperties(id);
 			owner = props.getProperty(ResourceProperties.PROP_CREATOR);
 		}
-		catch ( Exception e ) 
+		catch ( PermissionException e )
 		{
 			// PermissionException can be thrown if not RESOURCE_AUTH_READ
 			return false;
 		}
```
Comment (nicholaswilson100) [inline kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java:2529]: Is there any other way of doing this?  Empty catch blocks to control program flow...not ideal practice.  Bit of a rewrite probably?  That said, there are lots of examples of this so...

File: kernel-impl/src/test/java/org/sakaiproject/authz/impl/TwoFactorAuthenticationIntTest.java (hunk 1 above)
Comment (buckett) [inline kernel-impl/src/test/java/org/sakaiproject/authz/impl/TwoFactorAuthenticationIntTest.java:110]: Just means splitting up the method into multiple checks and doing the setup several times. SakaiKernelTestBase doesn't reset things between runs so you have to start being careful about using different resources or put it in different files which is very slow.

Comment (nicholaswilson100) [inline kernel-impl/src/test/java/org/sakaiproject/authz/impl/TwoFactorAuthenticationIntTest.java:110]: ok no worries

**Pull request merged**